src/runtime_debug.js

```diff
@@ -165,7 +165,7 @@ function dbg(...args) {
 #if ENVIRONMENT_MAY_BE_NODE && PTHREADS
   // Avoid using the console for debugging in multi-threaded node applications
   // See https://github.com/emscripten-core/emscripten/issues/14804
-  if (ENVIRONMENT_IS_NODE) {
+  if (ENVIRONMENT_IS_NODE && fs) {
```

Comment:
If we sometimes use `console.warn` below, before `fs` is set, then we might miss the warnings here? It might be safer to `require("fs").writeSync` explicitly as a workaround, as `dbg()` doesn't need to be fast anyhow?

Reply:
OTOH maybe that node bug has been fixed meanwhile and `console.warn` always works..?

Reply:
There is just one test that used `dbg()` in the `--pre-js`.

We could solve it by doing `require("fs").writeSync` alternatively yes.   I think this solution is perfectly fine though since using `dbg` in pre-js so rare that there is only one test in our entire suite that does it.

I think we will always prefer to using the direct stdout access when running on node.